Respect the default config and database path variables

horgh opened the pull request:
We set these in the Makefile and goreleaser configs, but as of the move
of them to the package, they weren't respected. This leads to ignoring
them when we shouldn't.

oschwald commented on bin/geoipupdate.c:
```diff
@@ -614,8 +660,11 @@ void md5hex_license_ipaddr(geoipupdate_s * gu, const char *client_ipaddr,
+         *client_ipaddr = NULL;
     char hex_digest[33], hex_digest2[33];
+    memset(hex_digest, 0, 33);
+    memset(hex_digest2, 0, 33);
```
You could just initialize these to the zero array:

```c
char hex_digest[33] = {0}, hex_digest2[33] = {0};
```

oschwald replied:
You could just initialize these:

```c
memset(hex_digest, 0, 33);
memset(hex_digest2, 0, 33);
```

Same comment below too.

horgh replied:
Good point. I was not aware that that syntax filled the entire array. Thanks for the tip!

horgh commented on bin/functions.c:
```diff
@@ -0,0 +1,401 @@
+// Return true if it is.
+bool is_valid_gzip_file(char const * const file)
+{
+    if (NULL == file || strlen(file) == 0) {
```
Wow, funny name for that style. I hadn't heard that name for it before!

Yeah, I'm not a fan of this style myself, but it appeared to be used in several locations in this code already. Although this code in general may not be something we want to be consistent with.

I'll adjust these.

This brings up another point: Should we collect some stuff like this into a C standards document? I know we talked about it not being so necessary yet, but some suggestions like this could help others. I was also thinking about bringing up use of `const`, both whether to use it where possible (as you can see I like it), and its position (I know some people prefer e.g. `const int` vs `int const`).

oschwald replied:
Starting a C style guide with basic stuff like this sounds good to me. We can start small and expand as stuff comes up.

horgh replied:
Cool. I'll write a draft later and send it to you!

horgh commented on bin/functions.c:
```diff
@@ -0,0 +1,401 @@
+{
+    if (NULL == file || strlen(file) == 0 ||
+            NULL == buf ||
+            bufsz == 0) {
```
Yep. I also updated README.dev to mention that!

horgh commented on bin/Makefile.am:
```diff
@@ -2,7 +2,22 @@ AM_CPPFLAGS = -I$(top_srcdir)/src -DSYSCONFDIR=\"$(sysconfdir)\" -DDATADIR=\"$(d
+
+EXTRA_DIST = geoipupdate-pureperl.pl
+
+TESTS = functions_test
```
Sounds good. How can I do that?

Related to this, would you prefer we used a more standard way of testing here? I just went with using `assert()` as you can see, which is fairly simple. I noticed that we use `libtap` in `libmaxminddb`. It would mean a dependency, but consistency might be worth it.

oschwald replied:
I am ok with the asserts. I don't think there is a reason to change the now.

oschwald commented on bin/Makefile.am:
```diff
@@ -1,4 +1,5 @@
-AM_CPPFLAGS = -I$(top_srcdir)/src -DSYSCONFDIR=\"$(sysconfdir)\" -DDATADIR=\"$(datadir)/GeoIP\" -Wall
+AM_CPPFLAGS = -I$(top_srcdir)/src -DSYSCONFDIR=\"$(sysconfdir)\" -DDATADIR=\"$(datadir)/GeoIP\" \
+			  -std=c99 -Wall -Wextra -Werror
```
I don't think we should enable `-Werror` by default as people compile on all sorts of compilers and some of them will emit warnings.

horgh replied:
Ah good point. Do you think we should drop `-std=c99` here too for similar reasons?

oschwald replied:
I believe we set `-std=c99` on `libmaxminddb` and have not run into any complaints yet. I think it is safe to leave that.

horgh commented:
I just rebased after the OSX merge.

horgh committed:
Respect the default config and database path variables

We set these in the Makefile and goreleaser configs, but as of the move
of them to the package, they weren't respected. This leads to ignoring
them when we shouldn't.

oschwald commented on CHANGELOG.md:
```diff
@@ -1,5 +1,14 @@
+
+* Respect the defaultConfigFile and defaultDatabaseDirectory variables in
+  the main package again. They were ignored in 4.1.0 through 4.1.4. If not
+  specified, these GitHub and PPA releases looked for the config in
```
"looked for .. the database directory in" is a bit awkward.

horgh replied:
True. I reworded it. Thank you.

horgh committed:
Reword changelog entry
